code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -71,6 +74,39 @@
 		can_bayonet = FALSE
 		update_icon()
 
+/obj/item/gun/ballistic/rifle/boltaction/attack_self(mob/user)
+	if(jammed)
+		if(prob(unjam_chance))
+			jammed = FALSE
+			unjam_chance = 10
+		else
+			unjam_chance += 10
+			to_chat(user, "<span class='warning'>[src] is jammed!</span>")
+			return FALSE
+	..()
+
+/obj/item/gun/ballistic/rifle/boltaction/process_fire(mob/user)
+	if(chambered.BB)
+		jamming_chance  += 5
+	if(prob(jamming_chance))
+		jammed = TRUE
+	..()
+
+/obj/item/gun/ballistic/rifle/boltaction/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/try_clean_weapon)
```

Comment:
are you sure that try_clean_weapon is the proc you want to be called here, and not check_attack_item or something? it looks like any item in the game could (in the current state of this PR) be used to attempt to clean a mosin with, unless I'm misreading/misunderstanding the code here

Reply:
did you check the proc for it

Reply:
Yeah you were right, sorry about that